### PR TITLE
Inventory interservice relations

### LINKED ISSUE #3040: Add support for inter-service relations
The goal of this ticket is to use the inter service relations in the web console. When a field is a reference to another service, the add and edit forms should:

- The web console should provide a select box (with search functionality) with all instances
- When a service identity attribute is provided, this value should be shown to the user.

This web-console should stay compatible with versions of inmanta/lsm that do not support this yet.

In the attribute list, it show should show the value (uuid or the identity) and link to that instance.

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

closes #3040 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] ~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/Slices/CreateInstance/UI/CreateInstance.test.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    apiHelper.resolve(
+      Either.right({ data: [ServiceInstance.a, ServiceInstance.b] })
+    );
+  });
+
+  const relationInputField = screen.getByPlaceholderText("Select an instance");
+  await userEvent.type(relationInputField, "ab");
+  expect(apiHelper.pendingRequests[0]).toEqual({
+    method: "GET",
+    url: `/lsm/v1/service_inventory/${InterServiceRelations.editable.entity_type}?include_deployment_progress=False&limit=100&filter.order_id=a`,
```
[the remark this answers is not in the document]
Comment (andraskvr) [reply src/Slices/CreateInstance/UI/CreateInstance.test.tsx:85]: Currently every change in the user input leads to a new request, so we have one where it's `a` and another where it's `ab`. I added another assertion to make this clearer.
We can also add some throttle/debounce to not fire a request after every change

File: src/Slices/CreateInstance/UI/CreateInstance.test.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  await act(async () => {
+    apiHelper.resolve(
+      Either.right({ data: [ServiceInstance.a, ServiceInstance.b] })
+    );
+  });
+  await act(async () => {
+    apiHelper.resolve(
+      Either.right({ data: [ServiceInstance.a, ServiceInstance.b] })
+    );
+  });
```
[the remark this answers is not in the document]
Comment (andraskvr) [reply src/Slices/CreateInstance/UI/CreateInstance.test.tsx:97]: See the other comment about the user input

File: src/Data/Managers/GetInstance/QueryManager.ts [hunk 3]
```diff
@@ -21,3 +21,21 @@ export class ServiceInstanceQueryManager extends QueryManager.ContinuousWithEnv<
     );
   }
 }
+
+export class GetServiceInstanceOneTimeQueryManager extends QueryManager.OneTimeWithEnv<"GetServiceInstance"> {
```
[the remark this answers is not in the document]
Comment (andraskvr) [reply src/Data/Managers/GetInstance/QueryManager.ts:25]: You can check the `docs/` for more details, but in a nutshell the difference is that a `Continuous` query gets executed regularly (every 5 seconds), while a `OneTime` query is executed only once by default. Both types are executed again if the parameters change

**Comment by andraskvr** [comment]
> I found this one a lot harder to review but I did my best, I think i'll pass a second time on the changes when the changes are done.

Yes, I'm sorry this turned out to be so long, it could have been split in at least 2 smaller ones

**Review by FloLey — APPROVED** [review]

**Pull request closed without merging**